clients/unshaded/src/main/java/tachyon/client/file/UnderStoreFileInStream.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import tachyon.client.ClientContext;
+import tachyon.client.block.BlockInStream;
+import tachyon.underfs.UnderFileSystem;
+
+/**
+ * This class provides a streaming API to read a fixed chunk from a file in the under storage
+ * system. The user may freely seek and skip within the fixed chunk of data. The under storage
+ * system read does not guarantee any locality and is dependent on the implementation of the under
+ * storage client.
+ */
+public class UnderStoreFileInStream extends BlockInStream {
```

Comment:
1 should this be `UnderStorageFileInStream`?
2 it is weird to see the `UnderStoreFileInStream` extending `BlockInStream`. Is it used as a block?